core/src/main/java/apoc/util/CompressionAlgo.java

```diff
@@ -39,18 +36,18 @@
         this.decompressor = decompressor;
     }
 
-    public List<Long> compress(String string, Charset charset) throws Exception {
+    public byte[] compress(String string, Charset charset) throws Exception {
         Constructor<?> constructor = compressor.getConstructor(OutputStream.class);
         try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
             try (OutputStream outputStream = (OutputStream) constructor.newInstance((OutputStream) stream)) {
```

Comment:
nit: you can write:
```java
try (ByteArrayOutputStream stream = new ByteArrayOutputStream();
     OutputStream outputStream = (OutputStream) constructor.newInstance((OutputStream) stream))) {
       // [...]
}
```

Reply:
I might be wrong, I think I cannot combine the two try-with-resources because in the inner try-with-res there is only `outputStream.write(string.getBytes(charset));` and the `return stream.toByteArray();` is out of this. If i join them, the function doesn't work.

Reply:
I could be wrong, but i think I can't combine the two try-with-res, because the inner try have only `outputStream.write(string.getBytes(charset));` and out of it there is the `return stream.toByteArray();`.
if I join them, the function does not work.

Reply:
Ah, you mean `outputStream` needs to be closed before `stream.toByteArray` is called?
If so, we could call `outputStream.flush()` and still merge the two `try`s, but that's probably not worth it.

Reply:
Yeah exactly. And also with `outputStream.flush()` after write() method, the compression doesn't work :(